vermouth/ffinput.py

```diff
@@ -359,7 +359,7 @@ def _dih_interactions(self, line, lineno=0, context_type=''):
           # [ impropers ] section.
 
         propers = []
-        impropers = []
+        impropers = context.interactions.get('impropers', [])
```

Comment:
I think this is OK. We should cache the already filtered impropers, which is implicitly done for dihedrals. I cannot remember which mechanism actually causes interactions of same atom names to be overwritten, but it is the default.